addons/sale_project/models/project.py

```diff
@@ -750,7 +750,7 @@ class ProjectTask(models.Model):
         copy=True, tracking=True, index='btree_not_null', recursive=True,
         compute='_compute_sale_line', store=True, readonly=False,
         domain="""[
-            '|', ('order_partner_id', 'child_of', partner_id if partner_id else []),
+            '|', ('order_partner_id.commercial_partner_id.id', 'parent_of', partner_id if partner_id else []),
```

Comment:
@xavierbol 
@mamatch 

Because is need endswith `.id`?. Time ago i remember a commit changed domain on some places
This domain cause performance issue
`('order_partner_id.commercial_partner_id.id', 'parent_of', partner_id if partner_id else [])`
This domain is better
`('order_partner_id.commercial_partner_id', 'parent_of', partner_id if partner_id else [])`

Reply:
I've tried without ``id`` but it doesn't work.

Reply:
You are right in general @celm1990 
adding the `.id` will force the ORM to do a LEFT JOIN / SUBSELECT that can be avoided entirely because it's stored on the model table, no need to go fetch it from the comodel.

BUT there is 1 **important** detail: using directly the field won't apply the `ir.rules` of the comodel in the query. So with an explicit `.id`, `ir.rules` on `res.partner` are applied inside the subqueries, which would lead to a change of behaviour, hence why @mamatch found it doesn't work without the explicite `.id`, because there are some rules inside base for `model_res_partner`.

Reply:
@pivi-odoo thanks a lot for clarify